### PR TITLE
Update docs remove extra returns from loss and extra args from callback

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: docs/src/examples/neural_ode/simplechains.md [hunk 1]
```diff
@@ -54,8 +54,7 @@ end
 function loss_neuralode(p)
     pred = predict_neuralode(p)
     loss = sum(abs2, data .- pred)
-    return loss, pred
-end
+    return loss
```
Comment (Vaibhavdixit02) [inline docs/src/examples/neural_ode/simplechains.md:57]:
```suggestion
    return loss
end
```

File: docs/src/examples/pde/pde_constrained.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ end
 ## Defining Loss function
 function loss(θ)
     pred = predict(θ)
-    return sum(abs2.(predict(θ) .- arr_sol)), pred # Mean squared error
+    return sum(abs2.(predict(θ) .- arr_sol)) # Mean squared error
 end
 
 l, pred = loss(ps)
```
Comment (Vaibhavdixit02) [inline docs/src/examples/pde/pde_constrained.md:73]:
```suggestion
l = loss(ps)
```

File: docs/src/examples/pde/pde_constrained.md [hunk 3]
```diff
@@ -67,25 +67,26 @@ end
...
+    pred = predict(st.u)
     append!(PRED, [pred])
     append!(LOSS, l)
-    append!(PARS, [θ])
+    append!(PARS, [st.u])
     false
 end
 
-cb(ps, loss(ps)...) # Testing callback function
+cb(ps, loss(ps)) # Testing callback function
```
Comment (Vaibhavdixit02) [inline docs/src/examples/pde/pde_constrained.md:89]:
```suggestion
cb((; u = ps), loss(ps)) # Testing callback function
```

File: docs/src/examples/ode/second_order_neural.md [hunk 4]
```diff
@@ -33,7 +33,7 @@ t = range(tspan[1], tspan[2], length = 20)
 model = Chain(Dense(2, 50, tanh), Dense(50, 2))
 ps, st = Lux.setup(Random.default_rng(), model)
 ps = ComponentArray(ps)
-model = Lux.Experimental.StatefulLuxLayer(model, ps, st)
+model = Lux.StatefulLuxLayer(model, ps, st)
```
Comment (Vaibhavdixit02) [inline docs/src/examples/ode/second_order_neural.md:36]:
```suggestion
model = Lux.StatefulLuxLayer{true}(model, ps, st)
```

**Comment by Vaibhavdixit02** [comment]
Not sure what the failure is now. It doesn't show any errors in the logs but a segmentation fault at the end?

**Comment by ChrisRackauckas** [comment]
Is it just the SDE control example? Did you isolate it locally?

File: docs/Project.toml [hunk 5]
```diff
@@ -45,7 +46,8 @@ Enzyme = "0.12, 0.13"
 Flux = "0.14"
 ForwardDiff = "0.10"
 IterTools = "1"
-Lux = "1"
+MLUtils = "0.4"
+Lux = "0.5.7, 1"
```
Comment (ChrisRackauckas) [inline docs/Project.toml:50]: why pull back Lux?

Comment (Vaibhavdixit02) [reply docs/Project.toml:50]: Hmm I think it showed up in a rebase conflict. Doesn't need to be held back

Comment (Vaibhavdixit02) [inline docs/Project.toml:50]:
```suggestion
Lux = "1"
```
